src/dataflow/src/render/reduce.rs

```diff
@@ -1063,6 +1066,135 @@ where
+    },
+    /// Accumulates the count of non-NULL values.
+    Count {
+        /// The number of non-NULL values observed.
+        non_nulls: isize,
+        /// The total number of values observed.
+        ///
+        /// This value is not actually observed directly by the code in this
+        /// module, but it serves to ensure that `is_zero()` returns false when
+        /// at least one non-NULL value has been observed. This is required to
+        /// produce an explicit `0` when a group contains zero elements;
+        /// otherwise differential would produce no results at all.
+        total: isize,
+    },
+    /// Accumulates simple numeric values.
+    SimpleNumber {
+        /// The sum of all non-NULL values observed.
+        sum: i128,
+        /// The number of non-NULL values observed.
+        count: isize,
+    },
+}
+
+impl Default for Accumulator {
+    fn default() -> Accumulator {
+        Accumulator::Count {
+            non_nulls: 0,
+            total: 0,
+        }
+    }
+}
+
+impl Semigroup for Accumulator {
+    fn is_zero(&self) -> bool {
+        match self {
+            Accumulator::Bool {
+                trues,
+                falses,
+                total,
+            } => trues.is_zero() && falses.is_zero() && total.is_zero(),
+            Accumulator::Count { non_nulls, total } => non_nulls.is_zero() && total.is_zero(),
+            Accumulator::SimpleNumber { sum, count } => sum.is_zero() && count.is_zero(),
+        }
+    }
+
+    fn plus_equals(&mut self, other: &Accumulator) {
+        match (self, other) {
+            (
+                Accumulator::Bool {
+                    trues,
+                    falses,
+                    total,
+                },
+                Accumulator::Bool {
+                    trues: other_trues,
+                    falses: other_falses,
+                    total: other_total,
+                },
+            ) => {
+                *trues += other_trues;
+                *falses += other_falses;
+                *total += other_total;
+            }
+            (
+                Accumulator::Count { non_nulls, total },
+                Accumulator::Count {
+                    non_nulls: other_non_nulls,
+                    total: other_total,
+                },
+            ) => {
+                *non_nulls += other_non_nulls;
+                *total += other_total;
+            }
+            (
+                Accumulator::SimpleNumber { sum, count },
+                Accumulator::SimpleNumber {
+                    sum: other_sum,
+                    count: other_count,
+                },
+            ) => {
+                *sum += other_sum;
+                *count += other_count;
+            }
+            (l, r) => unreachable!(
+                "Accumulator::plus_equals called with non-matching variants: {:?} vs {:?}",
+                l, r
+            ),
+        }
+    }
+}
+
+impl Multiply<isize> for Accumulator {
```

Comment:
I haven't looked at the new trait bounds but am a bit surprised you need multiply here because its only going in a reduce

Reply:
`explode` requires it. Because you might have `(record, time, 57)` and want to turn that in to something accumulable, and need to do something with the `57` (multiply it).